src/main/java/org/apache/sling/jcr/base/internal/ConfigurationUpdater.java

```diff
@@ -0,0 +1,212 @@
+    }
+
+    @Override
+    public void configurationEvent(final ConfigurationEvent event) {
+        if ( event.getType() == ConfigurationEvent.CM_UPDATED ) {
+            configurationUpdaterList.forEach(configurationUpdater -> {
+                configurationUpdater.updateProps(event);
+            });
+        }
+    }
+
+    /**
+     * Encode the value for the ldap filter: \, *, (, and ) should be escaped.
+     */
+    private static String encode(final String value) {
+        return value.replace("\\", "\\\\")
+                .replace("*", "\\*")
+                .replace("(", "\\(")
+                .replace(")", "\\)");
+    }
+    
+    protected abstract class Updater {
+
+        protected final String oldPid;
+        protected final String newPid;
+        protected final Map<String, String> propsToReplace;
+
+        public Updater(final String oldPid, final String newPid, final Map<String, String> propsToReplace) {
+            this.oldPid = oldPid;
+            this.newPid = newPid;
+            this.propsToReplace = propsToReplace;
+        }
+
+        protected abstract void updateProps(ConfigurationEvent event);
+
+        protected abstract void updateProps();
+
+        protected abstract Configuration createConfiguration(String oldPid) throws IOException;
+
+        /**
+         * Update a configuration
+         */
+        protected void updateProps(final Configuration sourceConfig, ConfigurationAdmin configurationAdmin) {
+            final Dictionary<String, Object> sourceProps = sourceConfig.getProperties();
+            final Dictionary<String, Object> targetProps = new Hashtable<>();
+            for(final String name : Collections.list(sourceProps.keys())) {
+                targetProps.put(this.propsToReplace.getOrDefault(name, name), sourceProps.get(name));
+            }
+            try {
+                final Configuration cfg = this.createConfiguration(sourceConfig.getPid());
+                if (cfg==null) return;
+                cfg.update(targetProps);
+                sourceConfig.delete();
+                logger.info("Updated configuration with PID {} to new configuration with PID {}. "+
+                "Please see https://sling.apache.org/documentation/the-sling-engine/service-authentication.html for more information.", 
+                sourceConfig.getPid(), cfg.getPid());
+            } catch (final IOException e) {
+                logger.warn("Failed to update configuration with PID {}", sourceConfig.getPid(), e);
+            }
+        }
+    }
+
+    private class PidConfigurationUpdater extends Updater {
+
+        public PidConfigurationUpdater(final String oldPid, final String newPid, final Map<String, String> propsToReplace) {
+            super(oldPid, newPid, propsToReplace);
+        }
+
+        @Override
+        protected void updateProps(final ConfigurationEvent event) {
+            if (this.oldPid.equals(event.getPid())) {
```

Comment:
Would it make sense to log a warning and instruct the user to use the new configuration? Also what if a user has both old and new configuration in place?

Reply:
> Good points. We log the situation when the configuration is actually converted with a link to the docs. Do you think we need more?

I think link to the docs in enough, but logging warning instead of info seems to be a better option. "Info" messages can typically be disregarded as they provide general information.

> If the new configuration exists, it will be overwritten by the old one and then the old one will be deleted.
We could handle this differently and not touch the new configuration but just log and delete the old one?

While both solutions appear to be viable options, IMO, it makes slightly more sense not to touch the new configuration. This approach aligns with our goal of encouraging users to migrate to the newer configurations. What do you think?